app/src/common/shared/com/igalia/wolvic/ui/widgets/menus/library/DownloadsContextMenuWidget.java

```diff
@@ -9,12 +9,9 @@
 
 public class DownloadsContextMenuWidget extends LibraryContextMenuWidget {
 
-    private boolean mCanCopyToContentUri;
-
     public DownloadsContextMenuWidget(Context aContext, LibraryContextMenuItem item,
                                       boolean canOpenWindows, boolean canCopyToContentUri) {
-        super(aContext, item, canOpenWindows, false);
-        mCanCopyToContentUri = canCopyToContentUri;
+        super(aContext, item, canOpenWindows, false, canCopyToContentUri);
```

Comment:
Hmm, I think the previous version was better because it was isolating the use case in the DownloadsContextMenuWidget. With this new version we're adding an extra parameter to several methods that won't ever use it. Why do you think it's better?

Reply:
Good suggestion, I will change the method's signature to use an enumerated type. This will also make the other related classes easier to read.

I had to change the previous code because it was not working as I expected: the menu items are created in the superclass' constructor, which runs before the subclass has stored the new parameter.

Reply:
This ended up being a bit more complex than expected, but I am reasonably satisfied with the result.

The subclasses of `LibraryContextMenuWidget` take `boolean` parameters only for the additional actions that are relevant for each one. These boolean parameters are then turned into an `EnumSet<Action>`, a set of actions.

(`EnumSet` is a specialised type of set for enum types that uses bit operations internally for good performance)

```java
EnumSet<Action> additionalActions = EnumSet.noneOf(Action.class);
if (canOpenWindows)
    additionalActions.add(Action.OPEN_WINDOW);
if (isBookmarked)
    additionalActions.add(Action.IS_BOOKMARKED);
```

This set is then used to instantiate the superclass and decide the right behaviour at each level of the class hyerarchy, so the code becomes a lot easier to read:

```java
if (additionalActions.contains(Action.OPEN_WINDOW)) {
    mItems.add(new MenuItem( ... ));
}
```